File: backend/backbone/src/main/java/com/party/backbone/websocket/model/GameType.java
package com.party.backbone.websocket.model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import lombok.Getter;

@Getter
public enum GameType {
	Reaction(45000),
	Clicker(30000),
	PerfectCircle(15000),
	Mugungwha(45000),
	Wirewalk(30000),
	Josephus(45000),
	Dye(30000),
	Knight(30000),
	GraphHigh(45000),
	Panopticon(30000),
	Dice(30000),
	ColorHunterG(25000),
	NumberSurvivor(70000),
	;
	private static final GameType[] GAME_TYPES = values();

	private final long duration;

	GameType(long duration) {
		this.duration = duration;
	}

	public static List<GameType> pickRandomList(int round) {
		if (round > GAME_TYPES.length) {
			throw new IllegalArgumentException("round cannot exceed available game types.");
		}

		List<GameType> list = new ArrayList<>(List.of(GAME_TYPES));
		Collections.shuffle(list);
		return list.subList(0, round);
	}
}

File: frontend/src/game/scenes/common/GameType.ts
// 게임 타입 정보를 따로 관리하는 컴포넌트(모듈)
export interface GameTypeInfo {
    name: string;
    key: string;
}

export const GAME_TYPES: GameTypeInfo[] = [
    { name: "반응속도 게임", key: "Reaction" },
    { name: "클리커", key: "Clicker" },
    { name: "원 그리기 게임", key: "PerfectCircle" },
    { name: "무궁화", key: "Mugungwha" },
    { name: "줄타기", key: "Wirewalk" },
    { name: "요세푸스", key: "Josephus" },
    { name: "염색", key: "Dye" },
    { name: "기사", key: "Knight" },
    { name: "숫자 생존게임", key: "NumberSurvivor" },
    { name: "판옵티콘", key: "Panopticon" },
    { name: "다이스", key: "Dice" },
];

